fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -447,6 +447,115 @@ public void commitTransaction(long dbId, long transactionId, List<TabletCommitIn
         // 6. update nextVersion because of the failure of persistent transaction resulting in error version
         updateCatalogAfterCommitted(transactionState, db);
         LOG.info("transaction:[{}] successfully committed", transactionState);
+
+        // 7. try publishing version as soon as possible
+        if (couldPublishTransaction(transactionState)) {
+            publishTransaction(transactionState, totalInvolvedBackends);
```

Comment:
If publish transaction here, there would be concurrent publish.
transactionState may be changed after `couldPublishTransaction`. Is it OK to call publishTrasaction concurrently?

Reply:
I think publishTrasaction won't concurrently.
`commitTransaction` need hold the db writeLock.
```
        db.writeLock();
        try {
            commitTransaction(db.getId(), transactionId, tabletCommitInfos, txnCommitAttachment);
        } finally {
            db.writeUnlock();
        }
```
 `partition.setNextVersion(partition.getNextVersion() + 1)` and  `                partition.updateVisibleVersionAndVersionHash(version, versionHash);` are both serially.